src/jit/hwintrinsicxarch.cpp

```diff
@@ -532,7 +592,10 @@ GenTree* Compiler::addRangeCheckIfNeeded(NamedIntrinsic intrinsic, GenTree* last
     assert(lastOp != nullptr);
     // Full-range imm-intrinsics do not need the range-check
     // because the imm-parameter of the intrinsic method is a byte.
-    if (mustExpand && !HWIntrinsicInfo::HasFullRangeImm(intrinsic) && HWIntrinsicInfo::isImmOp(intrinsic, lastOp))
+    // AVX2 Gather intrinsics no not need the range-check
+    // because their imm-parameter have discrete valid values that are handle by managed code
+    if (mustExpand && !HWIntrinsicInfo::isAVX2GatherIntrinsic(intrinsic) &&
```

Comment:
It seems like this should be the last check in the `&&` chain, since the other two are more likely to exit the chain first

Reply:
Agree, will fix